sdk/nodejs/docker.ts

```diff
@@ -264,15 +286,25 @@ async function buildAndPushImageWorkerAsync(
     if (pullFromCache) {
         const dockerBuild = <pulumi.UnwrappedObject<DockerBuild>>pathOrBuild;
         const cacheFromParam = (typeof dockerBuild.cacheFrom === "boolean" ? {} : dockerBuild.cacheFrom) || {};
-        cacheFrom = pullCacheAsync(baseImageName, cacheFromParam, repositoryUrl, logResource);
+
+        // pullFromCache is only true if repositoryUrl is present.
+        cacheFrom = pullCacheAsync(baseImageName, cacheFromParam, repositoryUrl!, logResource);
```

Comment:
just doc'ing why it's safe to do `repositoryUrl!` here. 

note: we could consider pulling 'pullCache' even higher and having a distinct set of steps at the top level of:

```
pullFromCache(...)
buildImage(...)
pushImage(...)
```